src/condor_startd.V6/command.cpp

```diff
@@ -1088,15 +1102,72 @@ request_claim( Resource* rip, Claim *claim, char* id, Stream* stream )
 
 	bool has_cp = false;
 	consumption_map_t consumption;
-	Claim* leftover_claim = NULL; 
+	Claim* pslot_claim = nullptr;
+
+	if (rip->is_partitionable_slot()) {
+		pslot_claim = rip->r_cur;
+	}
+
+	if (rip->is_partitionable_slot() && param_boolean("CLAIM_PARTITIONABLE_SLOT", false)) {
+		req_classad->LookupBool("_condor_CLAIM_PARTITIONABLE_SLOT", claim_pslot);
+		req_classad->LookupInteger("_condor_NUM_DYNAMIC_SLOTS", num_dslots);
+		req_classad->LookupInteger("_condor_PARTITIONABLE_SLOT_LEASE_TIME", pslot_claim_lease);
+		req_classad->LookupBool("condor_WANT_MATCHING", want_matching);
+	}
+
+	if (claim_pslot && rip->state() == claimed_state) {
+		rip->dprintf(D_ALWAYS, "Refusing claim of pslot that's already claimed\n");
+		refuse(stream);
+		ABORT;
+	}
+
+	if (claim_pslot) {
+		int max_pslot_claim_lease = param_integer("MAX_PARTITIONABLE_SLOT_LEASE_TIME", 3600);
+		if (pslot_claim_lease <= 0) {
+			pslot_claim_lease = max_pslot_claim_lease;
+		} else if (pslot_claim_lease > max_pslot_claim_lease) {
+			rip->dprintf(D_ALWAYS, "Refusing claim of pslot with lease time %d larger than max time %d\n", pslot_claim_lease, max_pslot_claim_lease);
```

Comment:
why fail rather than just bound the requested lease time?

Reply:
Because the design doc said so. I don't know if that was a strongly-held decision on ToddT's part.
If we allow the startd to bound the requested lease time, then we should have a way for the schedd to learn about it. Speaking of which, that's a useful thing to have in the pslot in general.

Reply:
...when the current claim will end, that is.

Reply:
Base on conversations today, I will change the code to bound and advertise the claim end time.